### PR TITLE
initial build for OPC UA Discovery module

### PR DESCRIPTION
**Pull request by mesudpasic**

### CODE REVIEW

File: Module.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+version: 1.0.0
+name: OPCUADiscovery
```
Comment (kubagrzelak) [inline Module.yaml]: please add spacebars `OPC UA Discovery`

File: Module.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+description: OPC UA Discovery looks for all available nodes onn OPC UA Server and forwards the payload to next module.
+icon: https://icons-020-demo.s3.eu-central-1.amazonaws.com/mi_OPCUA_input.png
```
Comment (kubagrzelak) [inline Module.yaml]: Same as in another module, it is `process` type module and you use `input` icon. Please ask Christina for process module icon for OPCUA

File: Module.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+type: process
+tags:
+  - OPC UA Discovery
+  - process
+categories:
+  - id: category
+    name: Process
```
Comment (kubagrzelak) [inline Module.yaml]: We don't have a category Process. I would suggest Policy category.

File: Module.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
+version: 1.0.0
+name: OPCUADiscovery
+description: OPC UA Discovery looks for all available nodes onn OPC UA Server and forwards the payload to next module.
+icon: https://icons-020-demo.s3.eu-central-1.amazonaws.com/mi_OPCUA_input.png
+type: process
```
Comment (kubagrzelak) [inline Module.yaml]: If this module subscribes to a server, why would it be of type process? If you build a pipeline with this module, do you MUST use another module before this one? If not, then it should be input module. If it is stand alone it seems that it does not need input right? If it does not need input from another module, it should be `input` type

File: README.md [hunk 5]
```diff
@@ -1,2 +1,60 @@
...
+
+### Set by the weeve Agent on the edge-node
+
+| Environment Variables | type   | Description               |
+| --------------------- | ------ | ------------------------- |
+| MODULE_NAME           | string | Name of the module        |
+| INGRESS_HOST          | string | Host where app is running |
+| INGRESS_PORT          | string | Port where app is running |
+
+## Output that is sent to next module looks like this
```
Comment (kubagrzelak) [inline README.md:41]: Is there no input example?

**Review by kubagrzelak — CHANGES_REQUESTED** [review]

File: Module.yaml (hunk 1 above)
Comment (mesudpasic) [reply Module.yaml]: done

File: Module.yaml (hunk 2 above)
Comment (mesudpasic) [reply Module.yaml]: we should use same icon for all these opc ua modules, kinda makes no sense anyway to have these names for icons anyway ? I'll request new icon, but I think I'll get the same.

File: Module.yaml (hunk 3 above)
Comment (mesudpasic) [reply Module.yaml]: updated. why do we have input, output and this one is called policy ?

File: README.md (hunk 5 above)
Comment (mesudpasic) [reply README.md:41]: not really.

File: Module.yaml (hunk 4 above)
Comment (mesudpasic) [reply Module.yaml]: ok, then it's input

File: Module.yaml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+      - no
+  - key:  OPC_UA_USERNAME
+    description: Username for authentication
+    value: ''
+    default: ''
+    type: string
+    name: OPC UA Username
+    dependencies:
+      - key: REQUIRES_AUTHENTICATION
+        value: yes
```
Comment (kubagrzelak) [inline Module.yaml]: dependencies field is not supported in V0 (we support in V1). This module is currently in V0 so please remove it.

File: Module.yaml [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  - key:  OPC_UA_PASSWORD
+    description: Password for authentication
+    value: ''
+    default: ''
+    type: string
+    name: OPC UA Password
+    dependencies:
+      - key: REQUIRES_AUTHENTICATION
+        value: yes
```
Comment (kubagrzelak) [inline Module.yaml]: Same

**Review by kubagrzelak — CHANGES_REQUESTED** [review]

File: Module.yaml (hunk 6 above)
Comment (mesudpasic) [reply Module.yaml]: removed.

File: Module.yaml (hunk 7 above)
Comment (mesudpasic) [reply Module.yaml]: removed.

File: Module.yaml (hunk 2 above)
Comment (mesudpasic) [reply Module.yaml]: module is changed to input based on our last call.

**Review by nithinsaii — APPROVED** [review]
Module.yaml looks good and container runs fine locally.

**Pull request merged**